ouroboros-consensus/docs/GenesisDesignOverview.md

```diff
@@ -0,0 +1,404 @@
+  we should see a portion of our recent chain when the density plummeted.
+
+We could instead collapse those two by defining the density up to the genesis
+block to be two times whatever the absolute cutoff is. So I'll only discuss the
+rejoining case from now on.
+
+## Balancing false positives and missed alarms
+
+The key challenge is false positives -- a node wrongly concluding it's
+eclipsed. This possibility is unavoidable, since we're working with partial
+information, much of which is random. As a drastic example: there is a
+probability that some epoch has no active slots, an empty leader schedule! That
+probability is astronomically small, but not zero. So we have to balance the
+sensitivity of our detection against its robustness, false alarms versus missed
+alarms. TODO also its promptness: we'd ideally detect an eclipse sooner rather
+than later... is a slow alarm just a partially/temporarily missed alarm?
+
+We can't judge that balance well without understanding the costs of each.
+
+* A missed alarm means some nodes were eclipsed and did not notice. In this
+  case the eclipsed nodes suffer the direct consequence, such as selling a
+  yacht without receiving payment on the honest chain.
+
+  The network as a whole wouldn't really notice the absence of a few nodes'
+  stake, which is inactive while they're eclipsed. On the other hand, it could
+  be distastrous if a large portion of the network were eclipsed simultaneously
+  for an appreciable duration/frequency/duty cycle (eg by a botnet saturating
+  our P2P peer selections).
+
+* A false alarm means that a node concluded it is eclipsed even though it
+  wasn't. The direct consequence is that the node leaves CaughtUp and goes back
+  to Connecting. If we put enough defensive mechanisms in place in Connecting
+  and Syncing (mostly timeouts), then such a (non-eclipsed!) node will take at
+  most a few minutes to return to CaughtUp (each false alarm).
+
+  A node in Connecting or Syncing is more likely to waste its leadership
+  opportunities by minting blocks that do not extend the best chain in the
+  network. This is because no growth (in Connecting) and Genesis prefix
+  selection (in Syncing) is slower to select that chain than is Praos chain
+  selection (in CaughtUp).
+
+  Again, the network as a whole wouldn't notice the absence of a few nodes'
+  stake, but it would be disastrous if the adversary could induce false alarms
+  in significant portions of the honest stake with appreciable
+  duration/frequency/duty cycle.
+
+## Why Genesis prefix selection can be slower than Praos chain selection
+
+It was decided the node should still mint blocks when in Connecting and
+Syncing. Any honest nodes minting stale blocks (creating several short forks)
+could slightly help the adversary create alternative chains, but the risk of
+many honest nodes skipping many slots they could have created a competitive is
```

Comment:
It seems there's something missing in this sentence :thinking:

Reply:
Chain?

Reply:
Done, thanks.